### PR TITLE
fix #80, more integer conversions

### LINKED ISSUE #80: convert / coerce n_Z into AbstractAlgebra.ZZ and/or Nemo.ZZ
`AbstractAlgebra.ZZ(Singular.ZZ(1))
`
fails with
`ERROR: MethodError: no method matching (::AbstractAlgebra.Integers{BigInt})(::n_Z)`

Neither do `convert(Int, Singular.ZZ(1))` and `convert(BigInt, Singular.ZZ(1))` work.

Similarly
`Nemo.ZZ(Singular.ZZ(1))`
fails with
`ERROR: MethodError: no method matching (::FlintIntegerRing)(::n_Z)`

I think such coercions / conversions would be a helpful feature.

### PR DESCRIPTION
**Pull request by rfourquet**

### CODE REVIEW

File: src/number/n_Z.jl [hunk 1]
```diff
@@ -357,6 +357,17 @@ end
 
 promote_rule(C::Type{n_Z}, ::Type{T}) where {T <: Integer} = n_Z
 
+BigInt(n::n_Z) = libSingular.n_GetMPZ(n.ptr, parent(n).ptr)
```
Comment (wbhart) [inline src/number/n_Z.jl:360]: I doubt you can have Julia clean up a BigInt that Singular allocated. This will probably result in crashes.

Comment (rfourquet) [reply src/number/n_Z.jl:360]: I just re-used the pre-existing definition for `convert(::Type{BigInt}, n::n_Z)`. Actually, in `n_GetMPZ`, the resulting `BigInt` is allocated on the Julia side (cf. https://github.com/oscar-system/Singular.jl/blob/796dbc9c0293b6dd004e17fb3898570b290d18b7/src/libsingular/coeffs.jl#L8).

**Comment by rfourquet** [comment]
The CI failures seem to be caused by the tests I added. I believe this will be fixed by #278.

**Comment by wbhart** [comment]
@rfourquet Could we rebase and then we'll see if #278 fixes this and we can get it merged.

**Pull request merged**